store/pruning/manager.go

```diff
@@ -0,0 +1,137 @@
+package pruning
+
+import (
+	"sync"
+
+	"cosmossdk.io/log"
+	"cosmossdk.io/store/v2"
+)
+
+// Manager is an abstraction to handle the pruning logic.
+type Manager struct {
+	stateStore      store.VersionedDatabase
+	stateCommitment store.Committer
+
+	logger         log.Logger
+	storeOpts      Options
+	commitmentOpts Options
+
+	chStore      chan struct{}
+	chCommitment chan struct{}
+
+	mtx sync.Mutex
+}
+
+// NewManager creates a new Manager instance.
+func NewManager(
+	stateStore store.VersionedDatabase,
+	stateCommitment store.Committer,
+	logger log.Logger,
+) *Manager {
+	return &Manager{
+		stateStore:      stateStore,
+		stateCommitment: stateCommitment,
+		logger:          logger,
+		storeOpts:       DefaultOptions(),
+		commitmentOpts:  DefaultOptions(),
+	}
+}
+
+// SetStoreOptions sets the store options.
+func (m *Manager) SetStoreOptions(opts Options) {
+	m.storeOpts = opts
+}
+
+// SetCommitmentOptions sets the commitment options.
+func (m *Manager) SetCommitmentOptions(opts Options) {
+	m.commitmentOpts = opts
+}
+
+// Start starts the manager.
+func (m *Manager) Start() {
+	m.mtx.Lock()
+	defer m.mtx.Unlock()
+
+	if !m.storeOpts.Sync {
+		m.chStore = make(chan struct{}, 1)
+		m.chStore <- struct{}{}
+	}
+	if !m.commitmentOpts.Sync {
+		m.chCommitment = make(chan struct{}, 1)
+		m.chCommitment <- struct{}{}
+	}
+}
+
+// Stop stops the manager and waits for all goroutines to finish.
+func (m *Manager) Stop() {
+	m.mtx.Lock()
+	defer m.mtx.Unlock()
+
+	if !m.storeOpts.Sync {
+		<-m.chStore
+		close(m.chStore)
+	}
+	if !m.commitmentOpts.Sync {
+		<-m.chCommitment
+		close(m.chCommitment)
+	}
+}
+
+// Prune prunes the state store and state commitment.
+// It will not block the caller and just check if pruning is needed.
+// If pruning is needed, it will spawn a goroutine to do the actual pruning.
+func (m *Manager) Prune(height uint64) {
+	m.mtx.Lock()
+	defer m.mtx.Unlock()
+
```

Comment:
We should return/no-op if the PM is not started. maybe add a `started bool` field.